packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  Address,
+  Customer,
+  PaymentSessionStatus,
+  ShippingMethod,
+} from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
+  shipping_address?: Address | null
+  shipping_methods?: ShippingMethod[]
```

Comment:
Both ShippingAddress and ShippingMethods seem out of place here. We want this to be only for processing payments of an amount in a given currency. 

Think it will be Checkout Provider API that would use things like Shipping details. Wdyt?